pkg/controller/tidbcluster/pod_control.go

```diff
@@ -458,6 +466,34 @@ func (c *PodController) syncTiDBPod(ctx context.Context, pod *corev1.Pod, tc *v1
 	return reconcile.Result{}, nil
 }
 
+func (c *PodController) cleanupLeaderEvictionAnnotations(pod *corev1.Pod, tc *v1alpha1.TidbCluster, annKeys []string) error {
+	for _, ann := range annKeys {
```

Comment:
Would be nice to be defensive
```suggestion
func (c *PodController) cleanupLeaderEvictionAnnotations(pod *corev1.Pod, tc *v1alpha1.TidbCluster, annKeys []string) error {
	if pod.Annotations == nil {
		return nil
	}
	for _, ann := range annKeys {
```